mmedit/apis/video_interpolation_inference.py

```diff
@@ -0,0 +1,140 @@
+            sequence, input_fps=0.0
+    """
+
+    device = next(model.parameters()).device  # model device
+
+    io_sequence = getattr(model.cfg['model'], 'io_sequence', [1, 1])
+
+    # build the data pipeline
+    if model.cfg.get('demo_pipeline', None):
+        test_pipeline = model.cfg.demo_pipeline
+    elif model.cfg.get('test_pipeline', None):
+        test_pipeline = model.cfg.test_pipeline
+    else:
+        test_pipeline = model.cfg.val_pipeline
+
+    # check if the input is a video
+    input_fps = 0.0
+    file_extension = os.path.splitext(input_dir)[1]
+    if file_extension in VIDEO_EXTENSIONS:
+        video_reader = mmcv.VideoReader(input_dir)
+        input_fps = video_reader.fps
+        images = []
+        # load the images
+        for img in video_reader[start_idx:end_idx]:
+            images.append(np.flip(img, axis=2))  # BGR --> RGB
+    else:
+        files = os.listdir(input_dir)
+        files = [osp.join(input_dir, f) for f in files]
+        files.sort()
+        files = files[start_idx:end_idx]
+        images = [read_image(f) for f in files]
+
+    data = []
+    before, after = io_sequence[0], io_sequence[1]
+
+    for i in range(0, len(images) - before - after + 1, before):
+        data.append(
+            dict(
+                inputs=images[i:i + before] +
+                images[i + before:i + before + after],
+                inputs_path=None,
+                key=input_dir))
+
+    # remove the data loading pipeline
+    tmp_pipeline = []
+    for pipeline in test_pipeline:
+        if pipeline['type'] not in [
+                'GenerateSegmentIndices', 'LoadImageFromFileList',
+                'LoadImageFromFile'
+        ]:
+            tmp_pipeline.append(pipeline)
+    test_pipeline = tmp_pipeline
+
+    # compose the pipeline
+    test_pipeline = Compose(test_pipeline)
+    data = [test_pipeline(d) for d in data]
+    data = scatter(collate(data, samples_per_gpu=1), [device])[0]['inputs']
+
+    # forward the model
+    result = []
+    index = 0
+    with torch.no_grad():
+        length = data.shape[0]
+        for start in range(0, length, batch_size):
+            end = start + batch_size
+            output = model(data[start:end], test_mode=True)['output'].cpu()
+            for j in range(output.shape[0]):
+                input_sequence = [
+                    np.flip(img, axis=2)
+                    for img in images[index:index + before]
+                ]
+                result.extend(input_sequence)
+                if len(output.shape) == 4:
+                    new_image = tensor2img(output[j])
+                    result.append(new_image)
+                else:
+                    for k in range(output.shape[1]):
+                        new_image = tensor2img(output[j][k])
+                        result.append(new_image)
+                if index + before + after >= len(images):
+                    result.extend(images[index + before:])
+                    break
```

Comment:
Similar to the above comment, there could be problems if `before` and `after` are not 1?